src/main/connections.c

```diff
@@ -4691,6 +4691,8 @@ readFixedString(Rconnection con, int len, int useBytes, Rboolean *warnOnNul)
 
 	p = buf = (char *) R_alloc(R_MB_CUR_MAX*len+1, sizeof(char));
 	memset(buf, 0, R_MB_CUR_MAX*len+1);
+	mbstate_t mb_st;
+	mbs_init(&mb_st);
 	for(i = 0; i < len; i++) {
```

Comment:
This is in a utf8 locale, so there should be no state, thus using local state should be safe no matter what.